activejob/test/cases/test_helper_test.rb

```diff
@@ -14,2054 +15,2138 @@
 require "jobs/multiple_kwargs_job"
 require "models/person"
 
-class EnqueuedJobsTest < ActiveJob::TestCase
-  def test_assert_enqueued_jobs
-    assert_nothing_raised do
-      assert_enqueued_jobs 1 do
-        HelloJob.perform_later("david")
-      end
+module DoNotPerformEnqueuedJobs
+  extend ActiveSupport::Concern
+
+  included do
+    setup do
+      # /rails/activejob/test/adapters/test.rb sets these configs to true, but
+      # in this specific case we want to test enqueueing behaviour.
+      @perform_enqueued_jobs = queue_adapter.perform_enqueued_jobs
+      @perform_enqueued_at_jobs = queue_adapter.perform_enqueued_at_jobs
+      queue_adapter.perform_enqueued_jobs = queue_adapter.perform_enqueued_at_jobs = false
+    end
+
+    teardown do
+      queue_adapter.perform_enqueued_jobs = @perform_enqueued_jobs
+      queue_adapter.perform_enqueued_at_jobs = @perform_enqueued_at_jobs
     end
   end
+end
+
+
+class EnqueuedJobsTest < ActiveJob::TestCase
+  if adapter_is?(:test)
```

Comment:
This is needed becuase you can't `skip` since https://github.com/rails/rails/pull/50188. Alternative would be to put a `force_skip` in a `setup` block.